nil-docs/zkllvm/getting-started/preparing-input-files.mdx

```diff
@@ -0,0 +1,191 @@
+| `"int"`, `"field"` and aggregated types that contain them | Scalar constants, strings (in case a number foes not fit within 64 bits), hexadecimal strings        |
+|                         `"curve"`                         | Arrays of numbers containing exactly two values                                                      |
+|             `"array"`, `"vector"`, `"struct"`             | Suitable aggregate types containing exactly the number of elements specified in the type description |
+
+## Extending keys in input files
+
+Except `"int"` and `"string"`, keys can be extended by specifying additional type details in angle brackets (`"<>"`) as shown in the below examples.
+
+- `"field<pallas_scalar>"`
+- `"vector<int>"`
+
+:::info[Aggregate types]
+
+Aggregate types (`"struct"`, `"array"`, and `"vector"`) are extended with the types of the elements they contain. In the case of `"array"` and `"vector"`, only a single type needs to be specified. For `"struct"`, it is possible to specify several types separated by a comma.
+
+:::
+
+:::info[Fields and curves]
+
+The `"field"` and `"curve"` keys can be extended by specifying the kind of their elements.
+
+:::
+
+The `"field"` key supports the following extensions.
+
+- `pallas_base`
+- `bls12381_base`
+- `ed25519_base`
+- `pallas_scalar`
+- `bls12381_scalar`
+- `ed25519_scalar`
+
+The `"curve"` key supports the following extensions.
+
+- `pallas`
+- `bls12381`
+- `ed25519`
+
+If an aggregate type is not extended, its elements must be specified as separate key-value pairs in the input file with keys containing their type descriptions.
+
+Consider the following example.
+
+<Tabs groupId='language'>
+  <TabItem value='cpp' label='C++'>
+
+    ```cpp
+    struct PublicInput {
+      int arg_one;
+      int arg_two;
+    };
+
+    [[circuit]] void circuit_func(PublicInput s,
+    int numbers[4]);
+    ```
+
+  </TabItem>
+  <TabItem value='rust' label='Rust'>
+
+    ```rust
+    struct PublicInput {  
+        arg_one: i32,  
+        arg_two: u32,  
+    }  
+
+    #[circuit]
+    fn circuit_func(s: PublicInput, numbers: [i32; 4]) {}
+    ```
+
+  </TabItem>
+</Tabs>
+
+There would be two ways to create a public input file for this example.
+
+<Tabs>
+  <TabItem value='fullSpec' label='With extensions'>
+    ```json
+    [
+      {"struct<int, int>": [10, 50]}, 
+      {"array<int>": [1, 2, 3, 4]}
+    ]
+    ```
+  </TabItem>
+  <TabItem value='lowSpec' label='Without extensions'>
+    ```json
+    [ {"struct": [
+        {"int": 10},
+        {"int": 50}
+      ]},
+      {"array": [
+        {"int": 1},
+        {"int": 2},
+        {"int": 3},
+        {"int": 4}
+      ]}
+    ]
+    ```
+  </TabItem>
+</Tabs>
+
+Similar rules would apply to the private input file.
+
+:::tip[Mixed representation]
+
+It is also possible to use mixed representations of aggregate types in the same input file as shown below.
```

Comment:
This input is not the same anymore with the previous, it would be correct for: 
```
struct PublicInput {
    std::array<int, 3> array_of_3;
    __zkllvm_field_pallas_base field_pallas;
};
```

Reply:
fair point. I did not plan on making it the same as the above examples are much simpler. But you're right, I've added new code showing the struct for which this input would be valid